protos/mission_server/mission_server.proto

```diff
@@ -0,0 +1,100 @@
+syntax = "proto3";
+
+package mavsdk.rpc.mission_server;
+
+import "mavsdk_options.proto";
+
+option java_package = "io.mavsdk.mission_server";
+option java_outer_classname = "MissionServerProto";
+
+// Enable waypoint missions.
+service MissionService {
+    /*
+     * Subscribe to when a new mission is uploaded (asynchronous).
+     */
+    rpc SubscribeIncomingMission(SubscribeIncomingMissionRequest) returns(stream IncomingMissionResponse) {}
+
+    /*
+     * Subscribe to when a new current item is set
+     */
+    rpc SubscribeCurrentItemChanged(SubscribeCurrentItemChangedRequest) returns(stream CurrentItemChangedResponse) {}
+
+    /*
+     *  Set Current item as completed
+     */
+    rpc SetCurrentItemComplete(SetCurrentItemCompleteRequest) returns(SetCurrentItemCompleteResponse) {}
+
+    /*
+     *  Subscribe when a MISSION_CLEAR_ALL is received
+     */
+    rpc SubscribeClearAll(SubscribeClearAllRequest) returns(stream ClearAllResponse) {}
+}
+
+message SubscribeIncomingMissionRequest {}
+message IncomingMissionResponse {
+    MissionResult mission_result = 1;
+    MissionPlan mission_plan = 2; // The mission plan
+}
+
+message SubscribeCurrentItemChangedRequest {}
+message CurrentItemChangedResponse {
+    MissionItem mission_item = 1;
+}
+
+message SubscribeClearAllRequest {}
+message ClearAllResponse {
+    uint32 clear_type = 1;
+}
+
+message SetCurrentItemCompleteRequest {}
+message SetCurrentItemCompleteResponse {}
+
+// Mission item exactly identical to MAVLink MISSION_ITEM_INT.
+message MissionItem {
```

Comment:
I suppose this should be a `MissionItemRaw` to keep the symmetry with the other plugins? Maybe even the plugin should be called `mission_raw_server`.

Reply:
Mission Raw plugin has the same type naming scheme here. I will change the plugin to `mission_raw_server` for parity.